packages/react-reconciler/src/ReactFiberBeginWork.js

```diff
@@ -3509,7 +3518,7 @@ function updateContextProvider(
     const providerPropTypes = workInProgress.type.propTypes;
 
     if (providerPropTypes) {
-      checkPropTypes(providerPropTypes, newProps, 'prop', 'Context.Provider');
+      checkPropTypes(providerPropTypes, nextProps, 'prop', 'Context.Provider');
```

Comment:
Why do we always check these in the reconciler instead of in createElement?

https://github.com/facebook/react/blob/16d053d592673dd5565d85109f259371b23f87e8/packages/react/src/ReactElementValidator.js#L224-L228